File: UI/src/microservices/requests/Request.ts
export interface IRequest {
  readonly requestId: string;
  readonly index: number;
  readonly x: number;
  readonly created: Date;
}

export class Request
  implements IRequest {
  readonly requestId: string;
  readonly index: number;
  readonly x: number;
  readonly created: Date;

  constructor(data: {requestId: string, index: number, x: number, created: any}) {
    this.requestId = data.requestId;
    this.index = data.index;
    this.x = data.x;
    this.created = new Date(data.created);
  }
}

File: RequestsMicroservice/src/RequestsMicroservice.Contracts/Requests/Request.cs
using System;

namespace RequestsMicroservice.Contracts.Requests
{
    public class Request
    {
        public Guid RequestId { get; set; }
        
        public int Index { get; set; }
        
        public double X { get; set; }

        public DateTime Created { get; set; }
    }
}